pkg/api/image/formatter.go

```diff
@@ -60,18 +62,76 @@ func (h UploadActionHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request
 	rw.WriteHeader(http.StatusOK)
 }
 
-func (h UploadActionHandler) do(rw http.ResponseWriter, req *http.Request) error {
+func (h ImageHandler) do(rw http.ResponseWriter, req *http.Request) error {
 	vars := mux.Vars(req)
-	action := vars["action"]
+	if req.Method == "GET" {
+		return h.doGet(vars["link"], rw, req)
+	} else if req.Method == "POST" {
+		return h.doPost(vars["action"], rw, req)
+	}
+
+	return apierror.NewAPIError(validation.InvalidAction, "Unsupported method")
+}
+
+func (h ImageHandler) doGet(link string, rw http.ResponseWriter, req *http.Request) error {
+	switch link {
+	case actionDownload:
+		return h.downloadImage(rw, req)
+	default:
+		return apierror.NewAPIError(validation.InvalidAction, "Unsupported GET action")
+	}
+}
+
+func (h ImageHandler) doPost(action string, rw http.ResponseWriter, req *http.Request) error {
 	switch action {
 	case actionUpload:
 		return h.uploadImage(rw, req)
 	default:
-		return apierror.NewAPIError(validation.InvalidAction, "Unsupported action")
+		return apierror.NewAPIError(validation.InvalidAction, "Unsupported POST action")
```

Comment:
add action string into error message will be better

Reply:
done